src/Sylius/Bundle/ApiBundle/Applicator/PaymentStateMachineTransitionApplicator.php

```diff
@@ -39,7 +41,11 @@ public function __construct(private StateMachineFactoryInterface|StateMachineInt
 
     public function complete(PaymentInterface $data): PaymentInterface
     {
-        $this->applyTransition($data, PaymentTransitions::TRANSITION_COMPLETE);
+        try {
+            $this->applyTransition($data, PaymentTransitions::TRANSITION_COMPLETE);
+        } catch (StateMachineExecutionException) {
+            throw new PaymentCompletionFailedException();
+        }
```

Comment:
Try-catch could be directly inside `applyTransition`, also why not check with `can` and throw only when it's false?